docs/asciidoc/static/deploying.asciidoc

```diff
@@ -0,0 +1,145 @@
+{logstash}pipeline.html[_processing pipeline_]. A given Logstash instance has a fixed pipeline constructed at startup, 
+based on the instance’s configuration file. You must specify an input plugin. Output defaults to `stdout`, and the 
+filtering section of the pipeline, which is discussed in the next section, is optional.
+
+image::images/deploy_1.png[]
+
+// [float]
+[[deploying-filter-threads]]
+=== Adding a Filter and Threads
+
+Log data is typically unstructured, often contains extraneous information that isn’t relevant to your use case, and 
+sometimes is missing relevant information that can be derived from the log contents. A 
+{logstash}filter-plugins.html[filter plugin] can handle these situations by parsing the log into fields, removing 
+unnecessary fields, and adding new information derived from the existing fields. As an example, geolocation information 
+can be derived from an IP address and added to the logs, or arbitrary text can be parsed and structured by the 
+{logstash}plugins-filters-grok.html[grok] filter.
+
+Depending on the amount of computation the filter plugin requires, and the volume of the logs being processed, adding a 
+filter plugin can have a significant effect on performance. The grok filter’s regular expression computation is 
+particularly resource-intensive. One way to better address this increased demand for computing resources is to use 
+parallel processing on multicore machines. Use the -w switch to set the number of execution threads for Logstash 
+filtering tasks. For example the bin/logstash -w 8 command uses eight different threads for filter processing.
+
+image::images/deploy_2.png[]
+
+// [float]
+[[deploying-logstash-forwarder]]
+=== Using Logstash Forwarder
+
+The https://github.com/elastic/logstash-forwarder[Logstash Forwarder] is a lightweight, resource-friendly tool written 
+in Go that collects logs from files on the server for processing on other machines. The Logstash Forwarder acts as a 
+client to a centralized Logstash instance, communicating using a secure protocol called Lumberjack. The Logstash 
+instance that acts as a server must be configured to use the {logstash}plugins-inputs-lumberjack.html[Lumberjack 
+input plugin].
+
+The Logstash Forwarder uses the computing resources of the machine hosting the source data, and the Lumberjack input 
+plugin minimizes the resource demands on the Logstash instance, making this architecture attractive for use cases with 
+resource constraints.
+
+image::images/deploy_3.png[]
+
+// [float]
+[[deploying-larger-cluster]]
+=== Scaling to a Larger Elasticsearch Cluster
+
+Typically, Logstash does not communicate with a single Elasticsearch node, but with a cluster made up of several nodes. 
+Logstash can use any of the protocols Elasticsearch makes available to move data into the cluster: 
+{guide}_transport_client_versus_node_client.html[HTTP, transport, or node]. 
```

Comment:
Lets add this sentence: In Logstash you can configure the different communication modes by setting the `protocol` setting in Elasticsearch output. Plus, lets link to this: https://www.elastic.co/guide/en/logstash/current/plugins-outputs-elasticsearch.html#plugins-outputs-elasticsearch-protocol